rocks/RocksIterator.java

```diff
@@ -81,18 +83,26 @@ public final boolean hasNext() {
             case EMPTY:
                 return fetchAndCheck();
             case INIT:
-                initalise();
-                return fetchAndCheck();
+                return initialise();
```

Comment:
I don't think we should merge `initialise()` and `fetchAndCheck()` together.

Reply:
By merging them together, we don't have to `synchronize` the `hasNext()` method because the inner methods that change the `state` are synchronized blocks. `synchronized` on `hasNext()` does have some performance implications (i tested) because it has to obtain a lock even when the iterator is just going to return `completed` or `fetched` I can have one call the other though and rename it